manifests/install.pp

```diff
@@ -7,6 +7,73 @@
   assert_private()
 
   case $facts['os']['family'] {
+    'Darwin': {
+      if $telegraf::manage_archive {
+        if $telegraf::manage_user {
+          group { $telegraf::config_file_group:
+            ensure => present,
+          }
+
+          user { $telegraf::config_file_owner:
+            ensure => present,
+            gid    => $telegraf::config_file_group,
+          }
+        }
+
+        $install_dir_deps = [
+          '/usr/local/bin',
+          '/usr/local/etc',
+          '/usr/local/opt',
+          '/usr/local/var',
+          '/usr/local/var/log',
+        ]
+
+        file { $install_dir_deps:
+          ensure => directory,
+        }
+
+        file { "${telegraf::archive_install_dir}-${telegraf::archive_version}":
+          ensure  => directory,
+          owner   => $telegraf::config_file_owner,
+          group   => $telegraf::config_file_group,
+          require => File[$install_dir_deps],
+        }
+
+        archive { "/tmp/telegraf-${telegraf::archive_version}.tar.gz":
+          ensure          => present,
+          extract         => true,
+          extract_path    => "${telegraf::archive_install_dir}-${telegraf::archive_version}",
+          extract_command => 'tar xfz %s --strip-components=2',
+          source          => "https://dl.influxdata.com/telegraf/releases/telegraf-${telegraf::archive_version}_darwin_amd64.tar.gz",
+          creates         => "${telegraf::archive_install_dir}-${$telegraf::archive_version}/usr/bin/telegraf",
+          user            => $telegraf::config_file_owner,
+          group           => $telegraf::config_file_group,
+          cleanup         => true,
+          require         => File["${telegraf::archive_install_dir}-${telegraf::archive_version}"],
+        }
+
+        file {
+          default:
+            ensure  => link,
+            require => Archive["/tmp/telegraf-${telegraf::archive_version}.tar.gz"],
+            ;
+          '/usr/local/bin/telegraf':
+            target  => "${telegraf::archive_install_dir}-${telegraf::archive_version}/usr/bin/telegraf",
+            ;
+          '/usr/local/etc/telegraf':
+            target  => "${telegraf::archive_install_dir}-${telegraf::archive_version}/etc/telegraf",
+            ;
+          '/usr/local/var/log/telegraf':
+            target  => "${telegraf::archive_install_dir}-${telegraf::archive_version}/var/log/telegraf",
+            ;
+        }
+
+        file { '/Library/LaunchDaemons/telegraf.plist':
+          ensure  => file,
+          content => epp('telegraf/telegraf.plist.epp', { 'config_file_owner' => $telegraf::config_file_owner, 'config_file_group' => $telegraf::config_file_group }),
```

Comment:
Minor request for readability:

```suggestion
          content => epp('telegraf/telegraf.plist.epp', {
            'config_file_owner' => $telegraf::config_file_owner,
            'config_file_group' => $telegraf::config_file_group,
          }),
```

Reply:
Sure thing, I almost did that, but for some reason VSCode started giving me fits about it.

Reply:
Not sure why the extra indentation does the trick 🤷